ui/raidboss/data/06-ew/raid/p4s.ts

```diff
@@ -142,6 +185,195 @@ const triggerSet: TriggerSet<Data> = {
+
+        // First Coils = Tethers later
+        if (roleTowers === 'dps') {
+          (data.tetherRole ??= []).push('healer');
+          data.tetherRole.push('tank');
+        } else {
+          (data.tetherRole ??= []).push('dps');
+        }
+
+        // For first coils, there are tower and tethers
+        if (data.tetherRole.includes(data.role))
+          return { ['alertText']: output.roleTethers!({ role: roles[roleOther] }) };
+        return { ['alertText']: output.roleTowers!({ role: roles[roleTowers] }) };
+      },
+      run: (data) => data.beloneCoilsTwo = true,
+    },
+    {
+      id: 'P4S Role Call',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: ['AF2', 'AF3'], capture: true }),
+      condition: Conditions.targetIsYou(),
+      alertText: (data, matches, output) => {
+        const debuffRole = (data.debuffRole ??= []).includes(data.role);
+        if (matches.effectId === 'AF2') {
+          // Call Pass Role Call if not in the debuff role
+          if (!debuffRole)
+            return output.passRoleCall!();
+          data.hasRoleCall = true;
+        }
+
+        // AF3 is obtained after passing Role Call (AF2)
+        if (matches.effectId === 'AF3')
+          data.hasRoleCall = false;
+      },
+      outputStrings: {
+        passRoleCall: {
+          en: 'Pass Role Call',
+        },
+      },
+    },
+    {
+      id: 'P4S Director\'s Belone',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '69E6', source: 'Hesperos', capture: false }),
+      netRegexDe: NetRegexes.ability({ id: '69E6', source: 'Hesperos', capture: false }),
+      netRegexFr: NetRegexes.ability({ id: '69E6', source: 'Hespéros', capture: false }),
+      netRegexJa: NetRegexes.ability({ id: '69E6', source: 'ヘスペロス', capture: false }),
+      // Delay callout until debuffs are out
+      delaySeconds: 1.4,
+      alertText: (data, _matches, output) => {
+        const debuffRole = (data.debuffRole ??= []).includes(data.role);
+        if (!data.hasRoleCall && debuffRole)
+          return output.text!();
+      },
+      outputStrings: {
+        text: {
+          en: 'Get Role Call',
+        },
+      },
+    },
+    {
+      id: 'P4S Inversive Chlamys',
+      // Possible a player still has not yet passed debuff
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '69ED', source: 'Hesperos', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ id: '69ED', source: 'Hesperos', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ id: '69ED', source: 'Hespéros', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ id: '69ED', source: 'ヘスペロス', capture: false }),
+      condition: (data) => !data.ignoreChlamys,
+      alertText: (data, _matches, output) => {
+        const roles: { [role: string]: string } = {
+          'dps': output.dps!(),
+          'tank/healer': output.tankHealer!(),
+          '???': output.unknown!(),
+        };
```

Comment:
I think this object is unnecessary, and you can just call `output.dps!()` instead of `roles['dps']` below.

Reply:
True. I don't know if this same replacement would work for the other section (bloodrake and coils) as the string is both used to compare data.role and to match to the roles output map.